dotCMS/src/main/java/com/dotcms/rendering/velocity/util/VelocityUtil.java

```diff
@@ -0,0 +1,607 @@
+					Logger.debug(
+					        VelocityUtil.class,
+							"The page is not available in language "
+									+ language.getId() + ". Just keep going.");
+
+					if(DEFAULT_PAGE_TO_DEFAULT_LANGUAGE) {
+						allDisplayLanguages.add(new DisplayedLanguage(language, true));
+					}
+
+					continue;
+				}
+			}
+			if(language.getId()==APILocator.getLanguageAPI().getDefaultLanguage().getId()) {
+				doesContentHaveDefaultLang = true;
+			}
+
+			languages.add(new DisplayedLanguage(language, false));
+
+			if(DEFAULT_PAGE_TO_DEFAULT_LANGUAGE) {
+				allDisplayLanguages.add(new DisplayedLanguage(language, false));
+			}
+		}
+
+		if(DEFAULT_PAGE_TO_DEFAULT_LANGUAGE && doesContentHaveDefaultLang){
+			return allDisplayLanguages;
+		}
+
+		return languages;
+	}
+
+	/**
+	 * This returns the proper ihtml page based on id, state and language
+	 * @param id
+	 * @param request
+	 * @param live
+	 * @return
+	 * @throws DotDataException
+	 * @throws DotSecurityException
+	 */
+	public static IHTMLPage getPage(Identifier id, HttpServletRequest request, boolean live, Context context) throws DotDataException, DotSecurityException{
+
+		long langId = getLanguageId(request);
+		request.setAttribute("idInode", String.valueOf(id.getId()));
+
+		IHTMLPage htmlPage;
+
+
+			Contentlet contentlet;
+
+			try{
+				contentlet = APILocator.getContentletAPI()
+								.findContentletByIdentifier(id.getId(),
+										live,
+										langId,
+										APILocator.getUserAPI().getSystemUser(), false);
+				htmlPage = APILocator.getHTMLPageAssetAPI().fromContentlet(contentlet);
+
+			} catch(DotStateException dse){
+				if(DEFAULT_PAGE_TO_DEFAULT_LANGUAGE && langId!= APILocator.getLanguageAPI().getDefaultLanguage().getId()){
+					contentlet = APILocator.getContentletAPI()
+									.findContentletByIdentifier(id.getId(),
+											live,
+											APILocator.getLanguageAPI().getDefaultLanguage().getId(),
+											APILocator.getUserAPI().getSystemUser(), false);
+					htmlPage = APILocator.getHTMLPageAssetAPI().fromContentlet(contentlet);
+				} else{
+					throw new DotDataException("Can't find content. Identifier: " + id.getId() + ", Live: " +live+ ", Lang: " + langId, dse);
+				}
+
+			}
+
+			if(UtilMethods.isSet(context)){
+				context.put("availablePageLangs", getAvailableContentPageLanguages(contentlet));
+			}
+		 
+
+		return htmlPage;
+	}
+
+	/**
+	 * Gets the Velocity Root Path. Looks for it on the Config, if not found the it get defaulted to /WEB-INF/velocity
+	 *
+	 * @return String
+	 */
+	public static String getVelocityRootPath() {
+		Logger.debug(VelocityUtil.class, "Fetching the velocity ROOT path...");
+
+		String velocityRootPath;
+
+		velocityRootPath = Config.getStringProperty("VELOCITY_ROOT", "/WEB-INF/velocity");
+		if (velocityRootPath.startsWith("/WEB-INF")) {
+			Logger.debug(VelocityUtil.class, "Velocity ROOT Path not found, defaulting it to '/WEB-INF/velocity'");
+			String startPath = velocityRootPath.substring(0, 8);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 8 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS109)